metagraph/src/graph/annotated_dbg.hpp

```diff
@@ -156,6 +156,10 @@ class AnnotatedDBG : public AnnotatedSequenceGraph {
                                      int32_t match_score = 1,
                                      int32_t mismatch_score = 2) const;
 
+    void call_annotated_rows(const std::vector<node_index> &rows,
+                             std::function<void(const std::string&)> callback_cell,
+                             std::function<void()> callback_row) const;
```

Comment:
What's the second callback supposed to be used for? It's not clear...

Reply:
I suggest to remove this function

Reply:
Thanks for the comments! I need to have access to  `annotator_` field and I can't implement it directly where needed.

Reply:
It is using 2 callback functions: one for doing a preprocessing per cell (i.e. get the taxid from label) and one for a final computation on an annotated row (i.e. get the LCA taxid). If you think that it would be more clear, I can reimplement it to use only one callback that receives the list of labels on one row (but this will use more memory -> store list of labels instead of list of integers)

Reply:
But do you need to have a non-const reference to the annotator? If not, you can just call `get_annotator().get_matrix()`

Reply:
Cool, I missed that `get_annotator()`. Did the changes now, thanks for the feedback